src/StructuredLogViewer/MainWindow.xaml.cs

```diff
@@ -659,31 +663,119 @@ private void Reload()
-                saveFileDialog.CheckFileExists = false;
-                saveFileDialog.OverwritePrompt = true;
-                saveFileDialog.ValidateNames = true;
-                var result = saveFileDialog.ShowDialog(this);
-                if (result != true)
+                if (
+                    !stringsToRedact.Any() &&
+                    !redactInputControl.RedactUsername &&
+                    !redactInputControl.RedactCommonCredentials)
                 {
+                    MessageBox.Show("No secrets to redact - no action will be performed");
                     return;
                 }
 
-                string newFilePath = saveFileDialog.FileName;
+                var progress = new BuildProgress();
+                progress.Progress.Updated += update =>
+                {
+                    Dispatcher.InvokeAsync(() =>
+                    {
+                        progress.Value = update.Ratio;
+                    }, DispatcherPriority.Background);
+                };
+                progress.ProgressText = "Performing the log redaction ...";
+                SetContent(progress);
+
+                string error = await System.Threading.Tasks.Task.Run(() =>
+                {
+                    try
+                    {
+                        BinlogRedactorOptions redactorOptions = new BinlogRedactorOptions(logFilePath)
+                        {
+                            OutputFileName = redactInputControl.DestinationFile,
+                            ProcessEmbeddedFiles = redactInputControl.RedactEmbeddedFiles,
+                            AutodetectUsername = redactInputControl.RedactUsername,
+                            AutodetectCommonPatterns = redactInputControl.RedactCommonCredentials,
+                            IdentifyReplacemenets = redactInputControl.DistinguishSecretsReplacements,
+                            TokensToRedact = stringsToRedact.ToArray(),
+                        };
+
+                        BinlogRedactor.RedactSecrets(
+                            redactorOptions,
+                            progress.Progress);
+                    }
+                    catch(Exception e)
+                    {
+                        return e.ToString();
+                    }
+
+                    return null;
+                });
+
+                if (!string.IsNullOrEmpty(error))
+                {
+                    MessageBox.Show($"Redaction failed:{Environment.NewLine}{error}");
+                    SetContent(currentBuild);
+                }
+                else if (string.IsNullOrEmpty(redactInputControl.DestinationFile))
+                {
+                    // Reload
+                    OpenLogFile(logFilePath);
+                }
+                else
+                {
+                    SetContent(currentBuild);
+                    AnnounceFileSaved(redactInputControl.DestinationFile);
+                }
+                

```

Comment:
no need for an empty line between two closing curlies. I should codify this in .editorconfig too.